index.bs

```diff
@@ -575,79 +575,225 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
+			<dd>
+			Creates a new {{ClipboardItem}} object. <var>items</var> are used to fill its MIME types and [=Promise=]s to [=Blob=]s or {{DOMString}}s corresponding to the MIME types, <var>options</var> can be used to fill its {{ClipboardItemOptions}},
+			as per the example below.
 
-		  readonly attribute PresentationStyle presentationStyle;
-		  readonly attribute long long lastModified;
-		  readonly attribute boolean delayed;
+				<pre class="example javascript" highlight=js>
+					const format1 = 'text/plain';
+					const promise_text_blob = Promise.resolve(new Blob(['hello'], {type: format1}));
+					const clipboardItemInput = new ClipboardItem(
+						{[format1]: promise_text_blob},
+						{presentationStyle: "unspecified"});
+				</pre>
 
-		  readonly attribute FrozenArray&lt;DOMString> types;
+			<dt><code><var>clipboardItem</var>.getType(<var>type</var>)</code>
+			<dd><p>Returns a [=Promise=] to the [=Blob=] corresponding to <var>type</var>.</p>
 
-		  Promise&lt;Blob> getType(DOMString type);
-		};
+			<dt><code><var>clipboardItem</var>.<var>types</var></code>
+			<dd><p>Returns the list of <var>types</var> contained in the <var>clipboardItem</var> object.
+
+		</dl>
+
+		<h4 id="clipboard-item">Clipboard Item</h4>
+
+		A [=clipboard item=] is conceptually data that the user has expressed a desire to make shareable by invoking a "cut" or "copy" command.
+
+		<p class=note>
+			For example, if a user copies a range of cells from a spreadsheet of a native application, it will result in one [=clipboard item=]. If a user copies a set of files from their desktop, that list of files will be represented by multiple [=clipboard item=]s.
+		</p>
+
+		Some platforms may support having more than one [=clipboard item=] at a time on the [=Clipboard=], while other platforms replace the previous [=clipboard item=] with the new one. 
+
+		A [=clipboard item=] has a <dfn>list of representations</dfn>, each <dfn>representation</dfn> with an associated [=mime type=] and <dfn>data</dfn>.
+
+		<p class=note>
+			In the example where the user copies a range of cells from a spreadsheet, it may be represented as an image (image/png), an HTML table (text/html), or plain text (text/plain).
+		</p>
+
+		Each of these [=mime type=]s describe a different [=representation=] of the same [=clipboard item=] at different levels of fidelity and make the [=clipboard item=] more consumable by target applications during paste.
+
+		<p class=note>
+			Making the range of cells available as an image will allow the user to paste the cells into a photo editing app, while the text/plain format can be used by text editor apps.
+		</p>
+
+		A [=clipboard item=] can also optionally have a <dfn>presentation style</dfn> that helps distinguish whether apps "pasting" a [=clipboard item=] should insert the contents of an appropriate [=representation=] inline at the point of paste or if it should be treated as an attachment.
```

Comment:
This sentence doesn't define the term `presentation style`. A term X defined via`<dfn>X</dfn>` is defined by the enclosing text, see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dfn. Shouldn't there only be one definition (which really is a definition) of the term "presentation style"? See also https://pr-preview.s3.amazonaws.com/w3c/clipboard-apis/pull/158.html#index-defined-here which shows multiple terms for "presentation style" with and without space and capital/non-capital letters.
Moreover, the presentation style isn't optional: if not overridden, it's "undefined" via `ClipboardItemOptions`.

Presumably, the marked sentence should be replaced by something like
`<p>A clipboard item has an associated <dfn ...>presentation style</dfn> which is a PresentationStyle.</p>`.

Reply:
Although your point about being more precise on the definition makes sense, I'll note that

> multiple terms for "presentation style" with and without space and capital/non-capital letters.

is correct. I.e., the spec needs to define the "presentation style" internal slot, and also define the `presentationStyle` IDL attribute and its getter steps.

Also, the current definition is good in that it explains what the "presentation style" internal slot actually does. Without that it'd be spec-conforming to implement presentation style but never pass it to the operation system or native apps.

Reply:
@domenic thanks for the feedback.

> the spec needs to define the "presentation style" internal slot, and also define the presentationStyle IDL attribute and its getter steps.

Makes sense.

> Also, the current definition is good in that it explains what the "presentation style" internal slot actually does. Without that it'd be spec-conforming to implement presentation style but never pass it to the operation system or native apps.

It seems a distinction of how a `ClipboardItem` is created is necessary.
For `clipboard.write()`, the presentation style is passed via JS when the `ClipboardItem` is constructed. Therefore, the `ClipboardItemOptions` dictionary should get a note for web-devs like
`Helps distinguish whether the ClipboardItem should insert the contents of an appropriate representation inline at the point of paste or if it should be treated as an attachment.`

For `clipboard.read()`, the presentation style information is obtained from the OS. Hence, the `read()` steps, defined in section 7.3.1, should retrieve that information from the OS, if it offers that. AFAIK, only Apple-OSs offer that. So that would have to be added to `Let data be a copy of the system clipboard data represented as clipboard items.`.

Further feedback appreciated.